From 6c3b032693acf86a894a8ea3a30c937a1d08ed7f Mon Sep 17 00:00:00 2001
From: aavit <eirik.aavitsland@digia.com>
Date: Tue, 12 Aug 2014 13:54:17 +0200
Subject: [PATCH 65/74] Fix QPainter::drawPolyline() painting errors with
 cosmetic pen

Task-number: QTBUG-31579
Change-Id: I8fd2c03ff9a22e4963bfcbcfe196ae4c61b9e10f
Reviewed-by: Gunnar Sletta <gunnar.sletta@jollamobile.com>
(cherry picked from qtbase/319cbb7597100f3b65792dc6a0ce2885ce6c0e8c)
Reviewed-by: Gunnar Sletta <gunnar@sletta.org>
---
 src/gui/painting/qcosmeticstroker.cpp | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/src/gui/painting/qcosmeticstroker.cpp b/src/gui/painting/qcosmeticstroker.cpp
index 5ca652e..1255a3b 100644
--- a/src/gui/painting/qcosmeticstroker.cpp
+++ b/src/gui/painting/qcosmeticstroker.cpp
@@ -533,8 +533,8 @@ void QCosmeticStroker::drawPath(const QVectorPath &path)
 
             QPointF p = QPointF(points[0], points[1]) * state->matrix;
             patternOffset = state->lastPen.dashOffset()*64;
-            lastPixel.x = -1;
-            lastPixel.y = -1;
+            lastPixel.x = INT_MIN;
+            lastPixel.y = INT_MIN;
 
             bool closed;
             const QPainterPath::ElementType *e = subPath(type, end, points, &closed);
@@ -588,8 +588,8 @@ void QCosmeticStroker::drawPath(const QVectorPath &path)
         QPointF p = QPointF(points[0], points[1]) * state->matrix;
         QPointF movedTo = p;
         patternOffset = state->lastPen.dashOffset()*64;
-        lastPixel.x = -1;
-        lastPixel.y = -1;
+        lastPixel.x = INT_MIN;
+        lastPixel.y = INT_MIN;
 
         const qreal *begin = points;
         const qreal *end = points + 2*path.elementCount();
-- 
1.9.3

